mitiq/pec/representations/biased_noise.py

```diff
@@ -0,0 +1,228 @@
+# Copyright (C) 2022 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Function to generate representations with biased noise."""
+
+from typing import List
+
+from cirq import (
+    Operation,
+    X,
+    Y,
+    Z,
+    Circuit,
+)
+
+from mitiq import QPROGRAM
+from mitiq.pec import OperationRepresentation, NoisyOperation
+from mitiq.interface import convert_to_mitiq, convert_from_mitiq
+
+
+def represent_operation_with_biased_noise(
+    ideal_operation: QPROGRAM,
+    epsilon: float,
+    eta: float,
+) -> OperationRepresentation:
+    r"""As described in [Strikis2021]_, this function maps an
+    ``ideal_operation`` :math:`\mathcal{U}` into its quasi-probability
+    representation, which is a linear combination of noisy implementable
+    operations :math:`\sum_\alpha \eta_{\alpha} \mathcal{O}_{\alpha}`.
+
+    This function assumes a combined depolarizing and dephasing noise model
+    with a bias factor :math:`\eta` and that the following noisy
+    operations are implementable
+    :math:`\mathcal{O}_{\alpha} = \mathcal{D} \circ \mathcal P_\alpha
+    \circ \mathcal{U}`, where :math:`\mathcal{U}` is the unitary associated
+    to the input ``ideal_operation`` acting on :math:`k` qubits,
+    :math:`\mathcal{P}_\alpha` is a Pauli operation and
+    :math:`\mathcal{D}(\rho) = (1 - \epsilon) \rho + \epsilon I/2^k` is a
+    combined (biased) depolarizing and dephasing channel.
```

Comment:
Here or before, we should give the definition of the biased noise channel (instead of `\mathcal{D}(\rho) = (1 - \epsilon) \rho + \epsilon I/2^k)` . At least we should give the formula for the single qubit biased noise channel.
Porbably it is also good to explain that, for multi-qubit operations, the noise channel is the tensor product of local single-qubit channels.

Reply:
Thanks for the reminder! I updated the formula in the docstring so that it's for the biased noise channel (on a single qubit) and added a line indicating "for multi-qubit operations, we use a noise channel that is the tensor product of the local single-qubit channels."